### PR TITLE
Kirkstone: fix halium armv7 targets

### PR DESCRIPTION
**Pull request by Tofee**

### CODE REVIEW

**Comment by shr-project** [comment]
The first patch shouldn't be needed.

**Comment by Tofee** [comment]
> The first patch shouldn't be needed.

the one about USE_X11? I remember cherry-picking it some time ago because of a build issue, but maybe that is indeed useless now. I'll retry without it.

**Comment by shr-project** [comment]
Yes, that one, it was replaced by:
2918e24c5 qtbase: add PACKAGECONFIG for egl_x11

**Comment by Tofee** [comment]
Thanks, I've removed commit from the PR

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+PV = "0.0.1+git${SRCPV}"
+SRCREV = "280047d9f5f159072a20f49a6d777a9f8da039f4"
+
+do_compile() {
+    cd ${S}
+    make test-shader LDFLAGS="${LDFLAGS}" CFLAGS="${CFLAGS}"
+}
+
+do_install() {
+    install -m 0644 -d ${D}/usr/bin
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:22]: bindir

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+SRC_URI = "git://github.com/Tofee/simple-shader.git;protocol=https;branch=main"
+S = "${WORKDIR}/git"
+
+PV = "0.0.1+git${SRCPV}"
+SRCREV = "280047d9f5f159072a20f49a6d777a9f8da039f4"
+
+do_compile() {
+    cd ${S}
+    make test-shader LDFLAGS="${LDFLAGS}" CFLAGS="${CFLAGS}"
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:18]: oe_runmake would respect PARALLEL_MAKE and EXTRA_OEMAKE (where you can pass LDFLAGS and CFLAGS).

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2019 Christophe Chapuis <chris.chapuis@gmail.com>
+
+DESCRIPTION = "Library of glib utilities."
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:3]: Use SUMMARY and this looks copy pasted from somewhere else :).

File: meta-luneos/recipes-qt/qt6/qtbase_git.bbappend [hunk 4]
```diff
@@ -106,6 +106,11 @@ SRC_URI:append = " \
     file://9901-Disable-Faux-bolding-in-Qts-FreeType-FontEngine.patch;maxver=6.3.0 \
 "
 
+# Upstream-Status: Inappropriate
```
Comment (shr-project) [inline meta-luneos/recipes-qt/qt6/qtbase_git.bbappend:109]: Upstream-Status belongs to the .patch files.

File: meta-luneos/recipes-qt/qt6/qtbase_git.bbappend [hunk 5]
```diff
@@ -106,6 +106,11 @@ SRC_URI:append = " \
...
+SRC_URI:append:hammerhead-halium = " file://0004-Patch-shaders-in-QRhiGles2-compileShader-to-fix-old-.patch"
+SRC_URI:append:tenderloin-halium = " file://0004-Patch-shaders-in-QRhiGles2-compileShader-to-fix-old-.patch"
+SRC_URI:append:mako = " file://0004-Patch-shaders-in-QRhiGles2-compileShader-to-fix-old-.patch"
```
Comment (shr-project) [inline meta-luneos/recipes-qt/qt6/qtbase_git.bbappend:112]: This makes qtbase and everything it belongs on it effectively MACHINE_ARCH.

Comment (Tofee) [reply meta-luneos/recipes-qt/qt6/qtbase_git.bbappend:112]: Would
``` SRC_URI:append:cortexa8t2hf-neon-halium ```
be the right approach?

Comment (shr-project) [reply meta-luneos/recipes-qt/qt6/qtbase_git.bbappend:112]: TUNE_PKGARCH (cortexa8t2hf-neon-halium) isn't an OVERRIDE:
```
# $OVERRIDES [4 operations]
#   set /OE/build/luneos-kirkstone/webos-ports/openembedded-core/meta/conf/bitbake.conf:801
#     "${TARGET_OS}:${TRANSLATED_TARGET_ARCH}:pn-${PN}:${MACHINEOVERRIDES}:${DISTROOVERRIDES}:${CLASSOVERRIDE}${LIBCOVERRIDE}:forcevariable"
#   append /OE/build/luneos-kirkstone/webos-ports/meta-webos-ports/meta-luneos/conf/distro/include/luneos.inc:106
#     [vardepsexclude] "WEBOS_EXTRA_MACHINEOVERRIDES"
#   set /OE/build/luneos-kirkstone/webos-ports/meta-webos-ports/meta-luneos/conf/distro/include/luneos.inc:206
#     [vardepsexclude] "MACHINE MACHINE_CLASS"
#   set /OE/build/luneos-kirkstone/webos-ports/openembedded-core/meta/conf/documentation.conf:304
#     [doc] "BitBake uses OVERRIDES to control what variables are overridden after BitBake parses recipes and configuration files."
# pre-expansion value:
#   "${TARGET_OS}:${TRANSLATED_TARGET_ARCH}:pn-${PN}:${MACHINEOVERRIDES}:${DISTROOVERRIDES}:${CLASSOVERRIDE}${LIBCOVERRIDE}:forcevariable"
OVERRIDES="linux-gnueabi:arm:pn-qtbase:armv7a:halium:hammerhead-halium:hammerhead-halium:luneos:class-target:libc-glibc:forcevariable"
```

Can it be applied to all halium builds?

Comment (Tofee) [reply meta-luneos/recipes-qt/qt6/qtbase_git.bbappend]: I applied it on armv7a , it makes more sense I think. But as this patch breaks tissot display, it needs some rework, so let's wait before merging anything

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+SRCREV = "d373356857a36707d740b65300a74ffaf74c14ae"
+
+do_compile() {
+    cd ${S}
+    oe_runmake test-shader
+}
+
+do_install() {
+    install -m 0644 -d ${D}${bindir}
+    make install DESTDIR=${D}${bindir}
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:23]: oe_runmake

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+PV = "0.0.1+git${SRCPV}"
+SRCREV = "d373356857a36707d740b65300a74ffaf74c14ae"
+
+do_compile() {
+    cd ${S}
+    oe_runmake test-shader
+}
+
+do_install() {
+    install -m 0644 -d ${D}${bindir}
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:22]: strange permissions for a directory, doesn't the Makefile create DESTDIR automatically?

File: meta-luneos/recipes-devtools/simple-shader/simple-shader.bb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# Copyright (c) 2019 Christophe Chapuis <chris.chapuis@gmail.com>
+
+DESCRIPTION = "Utility for testing GLES shaders on a device."
```
Comment (shr-project) [inline meta-luneos/recipes-devtools/simple-shader/simple-shader.bb:3]: SUMMARY

**Comment by Herrie82** [comment]
@Tofee Is this ready to merge now?

**Comment by Tofee** [comment]
@Herrie82 yes, it is

**Pull request merged**